pkg/apis/cartographer/v1alpha1/workload_source_printer.go

```diff
@@ -0,0 +1,190 @@
+	updatedRow := metav1beta1.TableRow{}
+
+	for _, r := range resources {
+		for _, o := range r.Outputs {
+			if r.Name == rq.resourceName && o.Name == rq.outputName {
+				imageRow.Cells = append(imageRow.Cells,
+					rq.imageHeader,
+					strings.Replace(o.Preview, "\n", "", -1),
+				)
+				updatedRow.Cells = append(updatedRow.Cells,
+					rq.updatedHeader,
+					printer.TimestampSince(o.LastTransitionTime, time.Now()),
+				)
+				rq.rows = append(rq.rows, imageRow, updatedRow)
+				break
+			}
+		}
+	}
+
+	return rq.rows
+}
+
+func WorkloadSourceImagePrinter(w io.Writer, workload *Workload) error {
+	printImageInfo := func(workload *Workload, printOpts table.PrintOptions) ([]metav1beta1.TableRow, error) {
+		sourceRow := metav1beta1.TableRow{}
+		sourceRow.Cells = append(sourceRow.Cells,
+			"Source:",
+			"Image",
+		)
+
+		rows := []metav1beta1.TableRow{sourceRow}
+
+		rq := resourcesQuery{
+			resourceName:  "image-provider",
+			outputName:    "image",
+			imageHeader:   "Image:",
+			updatedHeader: "Updated:",
+			rows:          rows,
+		}
+
+		rows = findResource(rq, workload.Status.Resources)
+
+		return rows, nil
+	}
+
+	tablePrinter := table.NewTablePrinter(table.PrintOptions{NoHeaders: true}).With(func(h table.PrintHandler) {
+		h.TableHandler(nil, printImageInfo)
+	})
+
+	return tablePrinter.PrintObj(workload, w)
+}
+
+func WorkloadSourceLocalImagePrinter(w io.Writer, workload *Workload) error {
+	printLocalSourceInfo := func(workload *Workload, printOpts table.PrintOptions) ([]metav1beta1.TableRow, error) {
+		sourceRow := metav1beta1.TableRow{}
+		sourceRow.Cells = append(sourceRow.Cells,
+			"Source:",
+			"Local source code",
+		)
+		rows := []metav1beta1.TableRow{sourceRow}
+
+		if workload.Spec.Source.Subpath != "" {
+			subPathRow := metav1beta1.TableRow{}
+			subPathRow.Cells = append(subPathRow.Cells,
+				"Sub-path:",
+				workload.Spec.Source.Subpath,
+			)
+			rows = append(rows, subPathRow)
+		}
+
+		rq := resourcesQuery{
+			resourceName:  "image-builder",
+			outputName:    "image",
+			imageHeader:   "Source-image:",
+			updatedHeader: "Updated:",
+			rows:          rows,
+		}
+
+		rows = findResource(rq, workload.Status.Resources)
+
+		return rows, nil
+	}
+
+	tablePrinter := table.NewTablePrinter(table.PrintOptions{NoHeaders: true}).With(func(h table.PrintHandler) {
+		h.TableHandler(nil, printLocalSourceInfo)
+	})
+
+	return tablePrinter.PrintObj(workload, w)
+}
+
+func WorkloadSourceGitPrinter(w io.Writer, workload *Workload) error {
+	printGitInfo := func(workload *Workload, printOpts table.PrintOptions) ([]metav1beta1.TableRow, error) {
+		sourceRow := metav1beta1.TableRow{}
+		sourceRow.Cells = append(sourceRow.Cells,
+			"Source:",
+			workload.Spec.Source.Git.URL,
+		)
+		rows := []metav1beta1.TableRow{sourceRow}
+
+		if workload.Spec.Source.Subpath != "" {
+			subPathRow := metav1beta1.TableRow{}
+			subPathRow.Cells = append(subPathRow.Cells,
+				"Sub-path:",
+				workload.Spec.Source.Subpath,
```

Comment:
```
				workload.Spec.Source.Subpath,
```
Is `Subpath` available in workload status?

Reply:
nope